### PR TITLE
check for monitor server start error

### PR DESCRIPTION
**Pull request by andyxning**
# Changes proposed in this pull request:

 - Add check for monitor server `Serve` returned error.
 -
 -

/cc @nats-io/core

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -1471,7 +1471,11 @@ func (s *Server) startMonitoring(secure bool) error {
 	s.mu.Unlock()
 
 	go func() {
-		srv.Serve(httpListener)
+		err := srv.Serve(httpListener)
+		if err != nil {
+			s.Errorf("Start %s monitor on %s error: %v", monitorProtocol,
```
Comment (derekcollison) [inline server/server.go]: Should this be `s.Fatalf` to match other listeners like clients? 

Also maybe break out hp and make error consistent.

```go
	hp := net.JoinHostPort(opts.HTTPHost, strconv.Itoa(opts.HTTPPort))
	s.Fatalf("Error starting monitoring on port: %s, %q", hp, err)
```

```

Comment (andyxning) [reply server/server.go]: Actually, `hp` has been initialized above, thus we can reference it directly.

should the `err` be formatted with `%q` or `%v`?

Comment (derekcollison) [reply server/server.go]: I usually use %v.

File: server/server.go [hunk 2]
```diff
@@ -1471,7 +1471,10 @@ func (s *Server) startMonitoring(secure bool) error {
 	s.mu.Unlock()
 
 	go func() {
-		srv.Serve(httpListener)
+		err := srv.Serve(httpListener)
```
Comment (derekcollison) [inline server/server.go]: Just combine?

```go
if err := srv.Serve(httpListener); err != nil {
```

File: server/server_test.go [hunk 3]
```diff
@@ -1162,6 +1162,8 @@ func TestConnectErrorReports(t *testing.T) {
 
 	remoteURLs := RoutesFromStr("nats://127.0.0.1:1234")
 
+	s.Shutdown()
```
Comment (derekcollison) [inline server/server_test.go]: This is deferred from above, why add here? Was test failing?

File: server/server.go (hunk 2 above)
Comment (andyxning) [reply server/server.go]: Done.

File: server/server_test.go (hunk 3 above)
Comment (andyxning) [reply server/server_test.go]: Yes, tests failed. But i have figured it out. We should only `Fatalf` when not in a `shutdown` stage.

**Comment by andyxning** [comment]
@derekcollison Done. PTAL.

**Pull request merged**